include/lcp-category.php

```diff
@@ -190,25 +189,46 @@ public function current_category($mode){
       if ( is_singular() || in_the_loop() ) {
         $categories = get_the_category($post->ID);
       }
+
+      $exclude = $this->excluded_ids($ids);
+
       if ( !empty($categories) ){
         $cats = array_map(function($cat) {
           return $cat->cat_ID;
         }, $categories);
-        // AND relationship
-        if ('all' === $mode) return $cats;
+        // AND relationship - do nothing, $cats is what we need
+        // if ('all' === $mode) {
+        //   return $cats;
+        // }
         // OR relationship, default
-        if ('yes' === $mode || '' === $mode) return implode(',', $cats);
+        if ('yes' === $mode || '' === $mode) {
+          $cats = implode(',', $cats);
+        }
         // Exclude current categories
-        if ('other' === $mode) return implode(',', array_map(function($cat) {
-          return "-$cat";
-        }, $cats));
+        if ('other' === $mode) {
+          $cats = implode(',', array_map(function($cat) {
+            return "-$cat";
+          }, $cats));
+        }
       } else {
-        return [0]; // workaround to display no posts
+        $cats = [0]; // workaround to display no posts
       }
     }
-    return $category->cat_ID;
```

Comment:
Just removing it would break current category detection on pages where category is set as a query var.

Maybe change  lines 181 - 192 to something like this?
```php
$category = get_category( get_query_var( 'cat' ) );
if( isset( $category->errors ) && $category->errors["invalid_term"][0] == __("Empty Term.") ){
  $cats = $category->cat_ID;
   /* Since WP 4.9 global $post is nullified in text widgets
    * when is_singular() is false.
    *
    * Added in_the_loop check to make the shortcode work
    * in posts listed in archives and home page (#358).
    */
} else if ( is_singular() || in_the_loop() ) {
  global $post;
  $categories = get_the_category($post->ID);
}
```
and adapt the code that follows to this new brace structure?